pkg/controller/utils/datadog/metrics_forwarder.go

```diff
@@ -64,28 +73,37 @@ type metricsForwarder struct {
 	globalTags          []string
 	tags                []string
 	stopChan            chan struct{}
+	errorChan           chan error
+	eventChan           chan Event
+	lastReconcileErr    error
 	namespacedName      types.NamespacedName
```

Comment:
can be replaced by `MonitoredObject`

Reply:
it's needed by the k8s client to `Get` CR object